prinz/src/main/scala/pl/touk/nussknacker/prinz/model/ModelSignature.scala

```diff
@@ -1,34 +1,48 @@
 package pl.touk.nussknacker.prinz.model
 
-import pl.touk.nussknacker.engine.api.typed.typing.TypingResult
+import pl.touk.nussknacker.engine.api.typed.typing.{TypedObjectTypingResult, TypingResult}
 import pl.touk.nussknacker.prinz.util.exceptions.Assertions.assertIllegal
 
-class ModelSignature private(signatureInputs: List[(SignatureName, SignatureType)], signatureOutputs: List[SignatureType]) {
+class ModelSignature private(signatureInputs: List[SignatureField], signatureOutputs: List[SignatureField]) {
 
-  private val signatureMap = signatureInputs.groupBy(_._1).mapValues(_.head._2)
+  private val signatureInputMap = signatureInputs.groupBy(_.signatureName).mapValues(_.head.signatureType)
 
-  def getSignatureNames: List[SignatureName] = signatureMap.keys.toList
+  private val signatureOutputMap = signatureOutputs.groupBy(_.signatureName).mapValues(_.head.signatureType)
 
-  def getValueType(valueName: SignatureName): Option[SignatureType] = signatureMap.get(valueName)
+  def getInputDefinition: TypedObjectTypingResult =
+    TypedObjectTypingResult(signatureInputMap.map(kv => (kv._1.name, kv._2.typingResult)))
 
-  def getOutputType: List[SignatureType] = signatureOutputs
+  def getOutputDefinition: TypedObjectTypingResult =
+    TypedObjectTypingResult(signatureOutputMap.map(kv => (kv._1.name, kv._2.typingResult)))
 
-  def getInputDefinition: List[(SignatureName, SignatureType)] = signatureInputs
+  def getSignatureInputNames: List[SignatureName] = signatureInputMap.keys.toList
+
+  def getSignatureOutputNames: List[SignatureName] = signatureOutputMap.keys.toList
+
+  def getInputValueType(valueName: SignatureName): Option[SignatureType] = signatureInputMap.get(valueName)
+
+  def getOutputValueType(valueName: SignatureName): Option[SignatureType] = signatureOutputMap.get(valueName)
+
+  def getInputList: List[SignatureField] = signatureInputs
+
+  def getOutputList: List[SignatureField] = signatureOutputs
```

Comment:
I think it's no longer needed, let's remove these two methods. You can create in place of them some kind of mthods that could be called explicitly in PrinzEnricher, so for example getInputNames and toParameterDefinition